From ab6bce807bf03c0bf98a7669c9c6ab75761e7e5a Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 20 May 2020 11:04:48 +0300
Subject: [PATCH] [SW-9973] spi: enable partial flash protection

As a preparation for partial (per device) SPI flash protection, we
change the condition for detecting the protection bits and SRWD bit, so
it will work with the partial flash protection variants.
Also rename the PROT_MASK into ALL_PROT_MASK, so devices can override
the PROT_MASK with device specific protection mask.

Change-Id: Ia5f245fdffecfde748d57c7af5357a82ef244cc8
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/spi.c b/common/spi.c
index 3165103..e057c87 100644
--- a/common/spi.c
+++ b/common/spi.c
@@ -39,7 +39,13 @@
 #define READ_STATUS_SRWD		BIT(7)
 #define READ_STATUS_BP0_3		(READ_STATUS_BP0 | READ_STATUS_BP1 | \
 					READ_STATUS_BP2 | READ_STATUS_BP3)
-#define PROT_MASK			(READ_STATUS_BP0_3 | READ_STATUS_SRWD)
+#define ALL_PROT_MASK			(READ_STATUS_BP0_3 | READ_STATUS_SRWD)
+
+/* The PROT_MASK is meant to be overridden by device specific setting */
+#ifndef PROT_MASK
+#define PROT_MASK			ALL_PROT_MASK
+#endif
+
 #define frame_size_bits(b)		((b) << 16)
 
 static unsigned int spi_regs;
@@ -243,7 +249,7 @@
 	if (prot_flash) {
 		prot_flash = false;
 		sr = spi_prot();
-		if ((sr & PROT_MASK) == PROT_MASK)
+		if ((sr & READ_STATUS_BP0_3) && (sr & READ_STATUS_SRWD))
 			preboot_puts("Flash may be protected, depend on WP# signal\n");
 		else
 			preboot_puts("Flash is not protected\n");
